lib/schema_tools/modules/attributes.rb

```diff
@@ -116,6 +118,54 @@ def schema
           @schema
         end
 
+        private
+
+        def process_string_type(field_format, value)
+          if field_format == 'date'
+            Date.parse(value) # or be explicit? Date.strptime('2001-02-03', '%Y-%m-%d')
+          elsif field_format == 'date-time'
+            Time.parse(value) # vs Time.strptime
+          else
+           value.to_s
+          end
+        end
+
+        def process_object_type(field_name, value)
+          if nested_class(field_name)
+            nested_class(field_name).from_hash(value)
+          else
+            value
+          end
+        end
+
+        def process_array_type(field_name, value)
+          if nested_class(field_name.to_s.singularize)
+            value.map do |element|
+              nested_class(field_name.to_s.singularize).from_hash(element)
+            end
+          else
+            value
+          end
+        end
+
+        def nested_class(field_name)
+          klass = "#{base_class}::#{field_name.to_s.camelize}"
+          if is_a_defined_class?(klass)
+            klass.constantize
+          end
+        end
+
+        def base_class
+          self.to_s.deconstantize
+        end
+
+        # This is a little ugly, because we cannot use Object.const_get. That is because
+        # const_get works only if the class has already been autoloaded. Find a way to refactor
+        def is_a_defined_class?(klass)
```

Comment:
That's break things like `user_data` and will enforce having a different class for each object. Not sure if we want that